### ISSUE TITLE
Superdesk publisher and wordpress

### ISSUE THREAD

**Issue by d1mf** [superdesk/web-publisher#590]
Hello. I have superdesk publisher. I want to add my WordPress website to superdesk.  I see need auth key. Can you 
![screenshot_156](https://user-images.githubusercontent.com/10393074/43480907-38974cca-950d-11e8-8caf-490e804f206a.jpg)
help?

**Comment by d1mf** [comment]
![screenshot_158](https://user-images.githubusercontent.com/10393074/43487695-0c06d1ae-9520-11e8-99f8-df6b1c89826f.jpg)
Wordpress need installed to server superdesk pusblisher?

**Comment by takeit** [comment]
Hi @d1mf,

please, check [Setup Wordpress as a Publisher Output Channel guide](https://superdesk-publisher.readthedocs.io/en/latest/cookbooks/implementators/wordpress_output_channel.html?highlight=wordpress#setup-wordpress-as-a-publisher-output-channel).

**Comment by d1mf** [comment]
Hello. Yes, I check. I see tenant not working server( without a superdesk publisher) I send a screenshot
![screenshot_159](https://user-images.githubusercontent.com/10393074/43506522-012c8c94-9573-11e8-8cfb-8acd50894544.jpg)
This server installed WordPress + router.php

**Comment by ahilles107** [comment]
In domain name and subdomain name you need to provide vhost (new one) for current publisher instance - not wordpress instance. Remove this tenant (probably manually from database).

**Comment by d1mf** [comment]
I check. Now I have a new problem. This server superdesk publisher.
![screenshot_160](https://user-images.githubusercontent.com/10393074/43515694-db977638-958b-11e8-8b6a-830cb4d2d90e.jpg)

not load theme and api error 500
![screenshot_161](https://user-images.githubusercontent.com/10393074/43515715-ed27ce70-958b-11e8-976b-ad3687a24010.jpg)



2018/08/01 03:06:10 [error] 36053#36053: *19 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Symfony\Component\Config\Definition\Exception\InvalidTypeException: Invalid type for path "nelmio_cors.defaults.allow_origin.0". Expected scalar, but got array. in /var/www/publisher/vendor/symfony/symfony/src/Symfony/Component/Config/Definition/ScalarNode.php:36
Stack trace:
#0 /var/www/publisher/vendor/symfony/symfony/src/Symfony/Component/Config/Definition/BaseNode.php(279): Symfony\Component\Config\Definition\ScalarNode->validateType(Array)
#1 /var/www/publisher/vendor/symfony/symfony/src/Symfony/Component/Config/Definition/PrototypedArrayNode.php(277): Symfony\Component\Config\Definition\BaseNode->normalize(Array)
#2 /var/www/publisher/vendor/symfony/symfony/src/Symfony/Component/Config/Definition/BaseNode.php(282): Symfony\Component\Config\Definition\PrototypedArrayNode->normalizeValue(Array)
#3 /var/www/publisher/vendor/symfony/symfony/src/Symfony/Component/Config/Definition/ArrayNode.php(301): Symfony\Component\Config\Definition\BaseNode->normalize(Array)
#4" while reading response header from upstream, client: 192.168.1.80, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "192.168.1.12"

**Comment by ahilles107** [comment]
error is here:  Invalid type for path "nelmio_cors.defaults.allow_origin.0". Expected scalar, but got array. in your parameters.yml you neet ot provide string not array for `nelmio_cors.defaults.allow_origin`

**Comment by d1mf** [comment]
This my full config

parameters:
    env(DATABASE_HOST): 127.0.0.1
    env(DATABASE_PORT): null
    env(DATABASE_NAME): publisher
    env(DATABASE_USER): postgres
    env(DATABASE_PASSWORD): postgres
    env(DATABASE_SERVER_VERSION): 9
    mailer_transport: smtp
    mailer_host: 127.0.0.1
    mailer_user: null
    mailer_password: null
    env(SYMFONY_SECRET): SuperSecretTokenPleaseChangeIt
    swp_updater.version.class: SWP\Bundle\CoreBundle\Version\Version
    env(SWP_DOMAIN): 192.168.1.12
    cache_servers:
        - localhost
    doctrine_cache_driver: array
    sentry.dsn: false
    session_memcached_host: localhost
    session_memcached_port: 11211
    session_memcached_prefix: sess
    session_memcached_expire: 3600
    test_env: doctrine
    allow_origin_cors: ['*']
    superdesk_servers: 192.168.1.6
    env(ELASTICA_HOST): localhost
    env(ELASTICA_PORT): 9200
    env(ELASTICA_INDEX_NAME): swp_index
    env(RABBIT_MQ_HOST): 127.0.0.1
    env(RABBIT_MQ_PORT): 5672
    env(RABBIT_MQ_USER): test
    env(RABBIT_MQ_PASSWORD): test
    env(WEBSOCKET_PORT): 8080
    env(WEBSOCKET_HOST): 127.0.0.1
    env(PAYWALL_SERVER_URL): null
    env(PAYWALL_SERVER_USERNAME): null
    env(PAYWALL_SERVER_PASSWORD): null
    env(SUBSCRIPTIONS_CACHE_LIFETIME): 86400
    fixtures_type: '%env(FIXTURES_TYPE)%'
    env(FIXTURES_TYPE): dev

I not see value nelmio_cors.defaults.allow_origin

Need to add?

**Comment by ahilles107** [comment]
`allow_origin_cors: ['*']` should be `allow_origin_cors: '*'`

**Comment by d1mf** [comment]
the testing server I see a fix. I see message Tenant for host "192.168.1.12" could not be found!( this not critical - testing server). Ok

This message prod(live server)

2018/08/01 11:17:23 [error] 1846#1846: *44 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Doctrine\ORM\NonUniqueResultException: More than one result was found for query although one row or none was expected. in /var/www/publisher/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php:770
Stack trace:
#0 /var/www/publisher/src/SWP/Bundle/MultiTenancyBundle/Doctrine/ORM/TenantRepository.php(59): Doctrine\ORM\AbstractQuery->getOneOrNullResult()
#1 /var/www/publisher/src/SWP/Component/MultiTenancy/Resolver/TenantResolver.php(54): SWP\Bundle\MultiTenancyBundle\Doctrine\ORM\TenantRepository->findOneByDomain('88.208.192.142')
#2 /var/www/publisher/app/cache/prod/Container2u1kqzx/TenantResolver_98081b0.php(13): SWP\Component\MultiTenancy\Resolver\TenantResolver->resolve('88.208.192.142')
#3 /var/www/publisher/src/SWP/Bundle/CoreBundle/Context/CachedTenantContext.php(79): TenantResolver_98081b0->resolve('88.208.192.142')
#4 /var/www/publisher/src/SWP/Bundle/CoreBundle/Security/Storage/DynamicDomainSessionStorage.php(41): SWP\Bundle\CoreBundle\Context\CachedTenan" while reading response header from upstream, client: 77.123.36.142, server: , request: "POST /wp-admin/admin-ajax.php HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "88.208.192.142", referrer: "http://88.208.192.142/wp-admin/"

# This file is auto-generated during the composer install
parameters:
    env(DATABASE_HOST): localhost
    env(DATABASE_PORT): null
    env(DATABASE_NAME): publisher
    env(DATABASE_USER): postgres
    env(DATABASE_PASSWORD): postgres
    env(DATABASE_SERVER_VERSION): 9
    mailer_transport: smtp
    mailer_host: 127.0.0.1
    mailer_user: null
    mailer_password: null
    env(SYMFONY_SECRET): SuperSecretTokenPleaseChangeIt
    swp_updater.version.class: SWP\Bundle\CoreBundle\Version\Version
    env(SWP_DOMAIN): 88.208.192.142
    cache_servers:
        - localhost
    doctrine_cache_driver: array
    sentry.dsn: false
    session_memcached_host: localhost
    session_memcached_port: 11211
    session_memcached_prefix: sess
    session_memcached_expire: 3600
    test_env: doctrine
    allow_origin_cors: '*'
    superdesk_servers: 213.171.204.147
    env(ELASTICA_HOST): localhost
    env(ELASTICA_PORT): 9200
    env(ELASTICA_INDEX_NAME): swp_index
    env(RABBIT_MQ_HOST): 127.0.0.1
    env(RABBIT_MQ_PORT): 5672
    env(RABBIT_MQ_USER): test
    env(RABBIT_MQ_PASSWORD): test
    env(WEBSOCKET_PORT): 8080
    env(WEBSOCKET_HOST): localhost
    env(PAYWALL_SERVER_URL): null
    env(PAYWALL_SERVER_USERNAME): null
    env(PAYWALL_SERVER_PASSWORD): null
    env(SUBSCRIPTIONS_CACHE_LIFETIME): 86400
    fixtures_type: '%env(FIXTURES_TYPE)%'
    env(FIXTURES_TYPE): dev

same error 500

**Comment by d1mf** [comment]
I understand I just want to get the application working. Very difficult. Thanks for the answer, I'll check it out.

**Comment by d1mf** [comment]
Done. I reinstall DB. Now I see this messages( server superdesk publisher)

Tenant for host "88.208.192.142" could not be found!

**Comment by ahilles107** [comment]
Hey, message is clear. After db reinstall did you created tenant?

**Comment by d1mf** [comment]
Sorry. I wrong this ( I mean wrong ip address) 

SYMFONY_ENV=prod php app/console swp:tenant:create  88.208.192.142 Superdesk

I see the theme load.

Next step I want to add WordPress. and API down. I add WordPress using the website(GUI).

Or need using this manual https://www.superdesk.org/news/installing-superdesk-publisher?topic=publisher ( switch  superdesk publisher to WordPress?)

**Comment by d1mf** [comment]
I not known what remove two tenants in the database with this same data. what find tenants?

**Comment by ahilles107** [comment]
Output controll and wordpress support is in not released yet version. So it may break. We spotted in last days many ui issues related to output channels, so i woudn't recommend you using it for creating new tenant. If You use latest master (there is open PR with improvements for wordpress adapter) then using api should work. 

In general when you have publisher running, then follow this cookbook https://superdesk-publisher.readthedocs.io/en/latest/cookbooks/implementators/wordpress_output_channel.html. 

Working with not released publisher features require some technicall knowledge - espcialy databes and php. I can recommend You to wait for next stable release (1.4) - planned for September. Then you will can configure all of it with UI.

**Comment by d1mf** [comment]
Is it possible to configure superdesk publisher + WordPress? This my task. If this needs only terminal, I can.
Or need to wait for new version support WordPress?

**Comment by ahilles107** [comment]
It is still work in progress. Many things works already but we are polishing the rest. Final release should be in September.

**Comment by d1mf** [comment]
Ok. Issue You can close

**Comment by d1mf** [comment]
Close
